diff -up GameLift_12_22_2020/GameLift-SDK-Release-4.0.2/GameLift-Cpp-ServerSDK-3.4.1/gamelift-server-sdk/source/aws/gamelift/server/GameLiftServerState.cpp.fixme GameLift_12_22_2020/GameLift-SDK-Release-4.0.2/GameLift-Cpp-ServerSDK-3.4.1/gamelift-server-sdk/source/aws/gamelift/server/GameLiftServerState.cpp
--- GameLift_12_22_2020/GameLift-SDK-Release-4.0.2/GameLift-Cpp-ServerSDK-3.4.1/gamelift-server-sdk/source/aws/gamelift/server/GameLiftServerState.cpp.fixme	2021-05-04 10:58:29.909350051 -0400
+++ GameLift_12_22_2020/GameLift-SDK-Release-4.0.2/GameLift-Cpp-ServerSDK-3.4.1/gamelift-server-sdk/source/aws/gamelift/server/GameLiftServerState.cpp	2021-05-04 10:58:23.683259064 -0400
@@ -14,6 +14,7 @@
 #include <aws/gamelift/server/protocols/sdk.pb.h>
 #include <fstream>
 #include <iostream>
+#include <thread>
 
 using namespace Aws::GameLift;
 
