diff -up o3de/scripts/o3de/setup.py.fixme o3de/scripts/o3de/setup.py
--- o3de/scripts/o3de/setup.py.fixme	2021-07-13 21:45:29.371629680 +0000
+++ o3de/scripts/o3de/setup.py	2021-07-13 21:45:42.934157199 +0000
@@ -27,11 +27,5 @@ if __name__ == '__main__':
         version="1.0.0",
         description='O3DE editor Python bindings test tools',
         long_description=long_description,
-        packages=find_packages(where='o3de', exclude=['tests']),
-        install_requires=[
-        ],
-        tests_require=[
-        ],
-        entry_points={
-        },
+        packages=find_packages(exclude=['tests']),
     )
