diff -up o3de/Code/Tools/ProjectManager/CMakeLists.txt.python-hack o3de/Code/Tools/ProjectManager/CMakeLists.txt
--- o3de/Code/Tools/ProjectManager/CMakeLists.txt.python-hack	2021-06-23 14:09:22.762586716 +0000
+++ o3de/Code/Tools/ProjectManager/CMakeLists.txt	2021-06-23 14:09:45.162451137 +0000
@@ -14,7 +14,8 @@ if(NOT PAL_TRAIT_BUILD_HOST_TOOLS)
 endif()
 
 # This will set python_package_name to whatever the package 'Python' is associated with
-ly_get_package_association(Python python_package_name)
+# ly_get_package_association(Python python_package_name)
+set(python_package_name python${Python_VERSION})
 if (NOT python_package_name)
     set(python_package_name "python-no-package-assocation-found")
     message(WARNING "Python was not found in the package assocation list.  Did someone call ly_associate_package(xxxxxxx Python) ?")
diff -up o3de/Gems/EditorPythonBindings/Code/CMakeLists.txt.python-hack o3de/Gems/EditorPythonBindings/Code/CMakeLists.txt
--- o3de/Gems/EditorPythonBindings/Code/CMakeLists.txt.python-hack	2021-06-21 14:59:30.000000000 +0000
+++ o3de/Gems/EditorPythonBindings/Code/CMakeLists.txt	2021-06-23 13:58:57.506371196 +0000
@@ -14,7 +14,8 @@ if(NOT PAL_TRAIT_BUILD_HOST_TOOLS)
 endif()
 
 # This will set python_package_name to whatever the package 'Python' is associated with
-ly_get_package_association(Python python_package_name)
+#ly_get_package_association(Python python_package_name)
+set(python_package_name python${Python_VERSION})
 if (NOT python_package_name)
     set(python_package_name "python-no-package-assocation-found")
     message(WARNING "Python was not found in the package assocation list.  Did someone call ly_associate_package(xxxxxxx Python) ?")
