diff -up civetweb-1.13/resources/Makefile.in-lua.fix-lua civetweb-1.13/resources/Makefile.in-lua
--- civetweb-1.13/resources/Makefile.in-lua.fix-lua	2021-03-25 14:44:33.582280903 -0400
+++ civetweb-1.13/resources/Makefile.in-lua	2021-03-25 14:44:49.346364552 -0400
@@ -38,7 +38,7 @@ endif
 ifeq ($(WITH_LUA_VERSION), 504)
   $(info Lua: Using version 5.4.0)
   LUA_DIR = src/third_party/lua-5.4.0/src
-  LUA_SHARED_LIB_FLAG = -llua5.4
+  LUA_SHARED_LIB_FLAG = -llua
   LUA_CFLAGS = -DLUA_COMPAT_5_2 -DLUA_VERSION_MAKEFILE=504
   LUA_VERSION_KNOWN = 1
 endif
