diff -up assimp-5.0.1/doc/Doxyfile.in.cmakemacros assimp-5.0.1/doc/Doxyfile.in
--- assimp-5.0.1/doc/Doxyfile.in.cmakemacros	2021-04-14 08:07:00.299356342 -0400
+++ assimp-5.0.1/doc/Doxyfile.in	2021-04-14 08:07:42.137652918 -0400
@@ -664,7 +664,6 @@ WARN_LOGFILE           =
 
 INPUT                  = @doxy_main_page@ \
                          @PROJECT_SOURCE_DIR@ \
-                         @PROJECT_BINARY_DIR@ \
                          @PROJECT_SOURCE_DIR@/include/ \
                          @PROJECT_SOURCE_DIR@/doc/dox.h \
                          @PROJECT_SOURCE_DIR@/code/BaseImporter.h
@@ -725,7 +724,8 @@ RECURSIVE              = YES
 # Note that relative paths are relative to the directory from which doxygen is 
 # run.
 
-EXCLUDE                = irrXML.h
+EXCLUDE                = irrXML.h \
+                         @PROJECT_BINARY_DIR@
 
 # The EXCLUDE_SYMLINKS tag can be used to select whether or not files or 
 # directories that are symbolic links (a Unix file system feature) are excluded 
