diff -up assimp-5.0.1/port/PyAssimp/pyassimp/helper.py.pythonpath assimp-5.0.1/port/PyAssimp/pyassimp/helper.py
--- assimp-5.0.1/port/PyAssimp/pyassimp/helper.py.pythonpath	2021-04-14 07:53:53.327688996 -0400
+++ assimp-5.0.1/port/PyAssimp/pyassimp/helper.py	2021-04-14 07:55:40.535470118 -0400
@@ -25,6 +25,7 @@ additional_dirs, ext_whitelist = [],[]
 # depending on the platform we're running on.
 if os.name=='posix':
     additional_dirs.append('./')
+    additional_dirs.append('/usr/lib64/')
     additional_dirs.append('/usr/lib/')
     additional_dirs.append('/usr/lib/x86_64-linux-gnu/')
     additional_dirs.append('/usr/local/lib/')
@@ -44,6 +45,7 @@ if os.name=='posix':
     # currently there's always a symlink called
     # libassimp.so in /usr/local/lib.
     ext_whitelist.append('.so')
+    ext_whitelist.append('.5')
     # libassimp.dylib in /usr/local/lib
     ext_whitelist.append('.dylib')
 
